### PR TITLE
filterSpam

### PR DESCRIPTION
**Pull request by jsh3d**
Let's just say someone typed this out:
Ur short
Ur short
Ur short
Ur short

It would be replaced with:
Ur short x4

Thats it ill add a readme.md later!

### CODE REVIEW

File: src/plugins/filterSpam/index.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        }
+    ],
+    start() {
+        FluxDispatcher.subscribe("MESSAGE_CREATE", this.onMessageCreate);
+    },
+    stop() {
+        FluxDispatcher.unsubscribe("MESSAGE_CREATE", this.onMessageCreate);
+    },
+    onMessageCreate({ message }) {
+        const root = document.querySelector(`.message-2qnXI6[id='chat-messages-${message.id}']`);
```
Comment (cheesesamwich) [inline src/plugins/filterSpam/index.tsx:64]: HORROR

Comment (imjustprism) [reply src/plugins/filterSpam/index.tsx:64]: whats horror? its okay:

- direct dom man
- weirdo content filtering logic
- getModule, pardon?
- u should use type anno

Comment (cheesesamwich) [reply src/plugins/filterSpam/index.tsx:64]: Don't specify the class names like that, the latter parts change with every discord update. Even if they didn't and you use optimised selectors, there are still more efficient ways to do it other than dom manipulation. Patch the message component maybe? (There might be better ways, i'm not sure)

File: src/plugins/filterSpam/index.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
+import { Devs } from "@utils/constants";
+import definePlugin from "@utils/types";
+import { FluxDispatcher, React, getModule } from "@webpack/common";
+import { findInReactTree } from "@utils/ReactUtils";
```
Comment (Vendicated) [inline src/plugins/filterSpam/index.tsx:4]: ?

File: src/plugins/filterSpam/index.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+const Message = getModule(m => m?.type?.displayName === 'Message');
```
Comment (Vendicated) [inline src/plugins/filterSpam/index.tsx:6]: display names have been gone for ages

File: src/plugins/filterSpam/index.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        cleanedContent.push(prevLine);
+    }
+
+    message.content = cleanedContent.join("\n");
+
+    return React.createElement(Message, props);
+};
+
+FilterSpamMessage.type = Message.type;
+FilterSpamMessage.displayName = Message.displayName;
```
Comment (Vendicated) [inline src/plugins/filterSpam/index.tsx:42]: ?

File: src/plugins/filterSpam/index.tsx (hunk 2 above)
Comment (jsh3d) [reply src/plugins/filterSpam/index.tsx:4]: did i accidently import a wrong library or smt

File: src/plugins/filterSpam/index.tsx (hunk 1 above)
Comment (Vendicated) [inline src/plugins/filterSpam/index.tsx:64]: this is either not a real class or very outdated. also never hardcode classes

**Review by Vendicated — COMMENTED** [review]
did you use AI to write this? it smells a lot like it

multiple things you use don't even exist

this plugin also just doesn't seem very useful. Just block spammers?

File: src/plugins/filterSpam/index.tsx (hunk 1 above)
Comment (jsh3d) [reply src/plugins/filterSpam/index.tsx:64]: aight

File: src/plugins/filterSpam/index.tsx (hunk 3 above)
Comment (jsh3d) [reply src/plugins/filterSpam/index.tsx:6]: no, ok, and its pretty useful if ur friend dm spams you like 20 times ig

**Pull request closed without merging**